backend/coreapp/serializers.py

```diff
@@ -1,11 +1,41 @@
-from coreapp.models import Profile, Scratch
+from django.contrib.auth.models import User
 from rest_framework import serializers
+from typing import Optional, TYPE_CHECKING
 
-class ProfileSerializer(serializers.ModelSerializer[Profile]):
-    class Meta:
-        model = Profile
-        fields = ["id", "username", "name", "avatar_url"]
+from .models import Profile, Scratch
+from .github import GitHubUser
+from .middleware import Request
+
+def serialize_profile(request: Request, profile: Profile):
+    if profile.user is None:
+        return {
+            "is_you": profile == request.profile,
+            "is_anonymous": True,
+        }
+    else:
+        user = profile.user
+        github: Optional[GitHubUser] = GitHubUser.objects.filter(user=user).first()
+
+        return {
+            "is_you": user == request.user,
+            "is_anonymous": False,
+            "id": user.id,
+            "username": user.username,
+            "email": user.email,
+            "name": github.details().name if github else user.username,
+            "avatar_url": github.details().avatar_url if github else None,
+            "github_api_url": github.details().url if github else None,
+            "github_html_url": github.details().html_url if github else None,
```

Comment:
May want to do something like `github_details = github.details() if github else None`? ( in case caching is disabled, e.g. `API_CACHE_TIMEOUT = 0`)